prow/config/config.go

```diff
@@ -1293,26 +1425,27 @@ func defaultPresubmits(presubmits []Presubmit, c *Config, repo string) error {
 
 // defaultPostsubmits defaults the postsubmits for one repo
 func defaultPostsubmits(postsubmits []Postsubmit, c *Config, repo string) error {
+	c.defaultPostsubmitFields(postsubmits)
 	var errs []error
 	for idx, ps := range postsubmits {
 		setPostsubmitDecorationDefaults(c, &postsubmits[idx], repo)
 		if err := resolvePresets(ps.Name, ps.Labels, ps.Spec, c.Presets); err != nil {
 			errs = append(errs, err)
 		}
 	}
-	c.defaultPostsubmitFields(postsubmits)
 	if err := SetPostsubmitRegexes(postsubmits); err != nil {
 		errs = append(errs, fmt.Errorf("could not set regex: %v", err))
 	}
 	return utilerrors.NewAggregate(errs)
 }
 
-// defaultPeriodics defaults periodics
-func defaultPeriodics(periodics []Periodic, c *Config) error {
+// defaultPeriodics defaults c.Periodics
+func defaultPeriodics(c *Config) error {
+	c.defaultPeriodicFields(c.Periodics)
 	var errs []error
-	c.defaultPeriodicFields(periodics)
-	for _, periodic := range periodics {
-		if err := resolvePresets(periodic.Name, periodic.Labels, periodic.Spec, c.Presets); err != nil {
+	for i := range c.Periodics {
+		setPeriodicDecorationDefaults(c, &c.Periodics[i])
+		if err := resolvePresets(c.Periodics[i].Name, c.Periodics[i].Labels, c.Periodics[i].Spec, c.Presets); err != nil {
```

Comment:
What, this never worked? :D

Reply:
I think it worked. The `periodic.Spec` field is all that `resolvePresets` modifies and it is a pointer type. I only changed this function signature because `setPeriodicDecorationDefaults` needs it and I wanted them to share the for loop for conciseness.